backend/asmgen.ml

```diff
@@ -153,18 +154,24 @@ let compile_fundecl ~ppf_dump fd_cmm =
   Reg.reset();
   fd_cmm
   ++ Profile.record ~accumulate:true "selection" Selection.fundecl
+  ++ Compiler_hooks.execute_and_pipe Compiler_hooks.Mach_sel
   ++ pass_dump_if ppf_dump dump_selection "After instruction selection"
-  ++ Profile.record ~accumulate:true "comballoc" Comballoc.fundecl
+  ++ Compiler_hooks.execute_and_pipe Compiler_hooks.Mach_combine
   ++ pass_dump_if ppf_dump dump_combine "After allocation combining"
+  ++ Profile.record ~accumulate:true "comballoc" Comballoc.fundecl
```

Comment:
You've moved `dump_combine` before the corresponding pass; I imagine that's not on purpose ?

Reply:
That wasn't on purpose....